lib/cyclonedx/cocoapods/bom_builder.rb

```diff
@@ -127,7 +127,7 @@ def add_to_bom(xml, manifest_path, trim_strings_length = 0)
           xml_add_author(xml, trim_strings_length)
           xml.name_ name
           xml.version version.to_s
-          xml.description { xml.cdata description } unless description.nil?
+          xml.description { xml.cdata description.dump[1..-2] } unless description.nil?
```

Comment:
Might be nice to add a comment on the line above this something along the lines of

```
# Use `dump` to escape non-printing characters, then remove the starting and trailing double-quotes from `dump`.
```

Also, would be nice to have a unit test that sets up a pod with a description that contains a null byte in the middle so we can be sure this continues to solve the problem.